storm-core/src/clj/org/apache/storm/cluster_state/zookeeper_state_factory.clj

```diff
@@ -144,7 +145,7 @@
        (let [curator-listener (reify ConnectionStateListener
                                 (stateChanged
                                   [this client newState]
-                                  (.stateChanged listener client newState)))]
+                                  (.stateChanged listener (StormConnectionStateConverter/convert newState))))]
```

Comment:
Thanks for kind words. :) I hope that Storm will get rid of many kinds of type / signature / typo problems when ported to Java, static typed language.